File: 0724-find-pivot-index/0724-find-pivot-index.py
class Solution:
    def pivotIndex(self, nums: List[int]) -> int:
        rightSum = 0

        for i in range(len(nums)):
            rightSum += nums[i]

        leftSum = 0

        for i in range(len(nums)):
            if leftSum == rightSum - nums[i]:
                return i

            leftSum += nums[i]
            rightSum -= nums[i]

        return -1

File: 0724-find-pivot-index/0724-find-pivot-index.cpp
class Solution {
public:
    int pivotIndex(vector<int>& nums) {
        int totalSum = 0;

        for (int i = 0; i < nums.size(); i++){
            totalSum = totalSum + nums[i];
        }
        int leftSum = 0;
        
        for (int i = 0; i < nums.size(); i++){
            int rightSum = totalSum - leftSum - nums[i];
        

        if (leftSum == rightSum){
            return i;
        }

        leftSum += nums[i];
        
        }

        return -1;
    }
};